discord/ext/commands/cooldowns.py

```diff
@@ -127,6 +128,8 @@ def _bucket_key(self, msg):
             return ((msg.guild and msg.guild.id), msg.author.id)
         elif bucket_type is BucketType.category:
             return (msg.channel.category or msg.channel).id
+        elif bucket_type is BucketType.role:
+            return msg.author.top_role.id
```

Comment:
This breaks when it's in a DM context.

For some reason so does the `BucketType.category` one. I must have missed that one during review.

Reply:
Just want to know your thoughts on this, would checking if you're in a DM context and then returning the user id be a good fix for that bug?

Reply:
Sure

Reply:
Fixed with the latest commit 6c5fee0